Bukkit/src/main/java/com/plotsquared/bukkit/listener/EntityEventListener.java

```diff
@@ -310,16 +329,92 @@ public void onBigBoom(EntityExplodeEvent event) {
     @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
     public void onPeskyMobsChangeTheWorldLikeWTFEvent(EntityChangeBlockEvent event) {
         Entity e = event.getEntity();
-        if (!(e instanceof FallingBlock)) {
-            Location location = BukkitUtil.adapt(event.getBlock().getLocation());
-            PlotArea area = location.getPlotArea();
-            if (area != null) {
-                Plot plot = area.getOwnedPlot(location);
-                if (plot != null && !plot.getFlag(EntityChangeBlockFlag.class)) {
-                    plot.debug(e.getType() + " could not change block because entity-change-block = false");
+        Material type = event.getBlock().getType();
+        Location location = BukkitUtil.adapt(event.getBlock().getLocation());
+        if (!this.plotAreaManager.hasPlotArea(location.getWorldName())) {
+            return;
+        }
+        PlotArea area = location.getPlotArea();
+        if (area == null) {
+            return;
+        }
+        if (e instanceof FallingBlock) {
+            // allow falling blocks converting to blocks and vice versa
+            return;
+        } else if (e instanceof Boat) {
+            // allow boats destroying lily pads
+            if (type == Material.LILY_PAD) {
+                return;
+            }
+        } else if (e instanceof Player player) {
+            BukkitPlayer pp = BukkitUtil.adapt(player);
+            if (type.toString().equals("POWDER_SNOW")) {
+                // Burning player evaporating powder snow. Use same checks as
+                // trampling farmland
+                BlockType blockType = BukkitAdapter.asBlockType(type);
+                if (!this.eventDispatcher.checkPlayerBlockEvent(pp,
+                        PlayerBlockEventType.TRIGGER_PHYSICAL, location, blockType, true)) {
+                    event.setCancelled(true);
+                }
+                return;
+            } else {
+                // already handled by other flags (mainly the 'use' flag):
+                // - player tilting big dripleaf by standing on it
+                // - player picking glow berries from cave vine
+                // - player trampling farmland
+                // - player standing on or clicking redstone ore
+                return;
+            }
+        } else if (e instanceof Projectile entity) {
+            // Exact same as the ProjectileHitEvent listener, except that we let
+            // the entity-change-block determine what to do with shooters that
+            // aren't players and aren't blocks
+            Plot plot = area.getPlot(location);
+            ProjectileSource shooter = entity.getShooter();
+            if (shooter instanceof Player) {
+                PlotPlayer<?> pp = BukkitUtil.adapt((Player) shooter);
+                if (plot == null) {
+                    if (!Permissions.hasPermission(pp, Permission.PERMISSION_PROJECTILE_UNOWNED)) {
+                        entity.remove();
```

Comment:
Does it make sense to remove the entity? (also applies for the other spots, more of a general question)

Reply:
In my mind the projectile is 'hitting' the block to interact with it, so it makes most sense to me to be consistent with the behaviour of the ProjectileHitEvent listener. I don't feel particularly strongly about removing the projectile though.

Reply:
You're right, this is the more consistent way then.